Scylla Operator crashes if automaticOrphanedNodeCleanup is enabled

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Slack](https://scylladb-users-slackin.herokuapp.com/).
3. Did you already search the existing open issues for anything similar? -->

**Describe the bug**
Scylla Operator crashes if `automaticOrphanedNodeCleanup` is enabled in a ScyllaCluster. 

**To Reproduce**
1. Create Operator as described 
2. Create a ScyllaCluster with `automaticOprhanedNodeCleanup: true` 
3. Operator starts and keeps crashing

**Expected behavior**
Operator should not crash. 

**Logs**
```
"L":"DEBUG","T":"2021-05-11T11:28:48.990Z","N":"cluster-controller","M":"Cleanup: service list","cluster":"scylla/scyllacluster","resourceVersion":"251886","len":1,"items":[{"kind":"Service","apiVersion":"v1","metadata":{"name":"scyllacluster-eu-central-1-a-0","namespace":"scylla","uid":"45a46a29-729d-45fd-983b-2da5822a4e29","resourceVersion":"248723","creationTimestamp":"2021-05-11T11:08:16Z","labels":{"app":"scylla","app.kubernetes.io/managed-by":"scylla-operator","app.kubernetes.io/name":"scylla","scylla/cluster":"scyllacluster","scylla/datacenter":"eu-central-1","scylla/rack":"a","scylla/seed":""},"ownerReferences":[{"apiVersion":"scylla.scylladb.com/v1","kind":"ScyllaCluster","name":"scyllacluster","uid":"6eeac944-d794-4a3c-81fb-1af18b8c2b5d","controller":true,"blockOwnerDeletion":true}],"managedFields":[{"manager":"scylla-operator","operation":"Update","apiVersion":"v1","time":"2021-05-11T11:08:16Z","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:labels":{".":{},"f:app":{},"f:app.kubernetes.io/managed-by":{},"f:app.kubernetes.io/name":{},"f:scylla/cluster":{},"f:scylla/datacenter":{},"f:scylla/rack":{},"f:scylla/seed":{}},"f:ownerReferences":{".":{},"k:{\"uid\":\"6eeac944-d794-4a3c-81fb-1af18b8c2b5d\"}":{".":{},"f:apiVersion":{},"f:blockOwnerDeletion":{},"f:controller":{},"f:kind":{},"f:name":{},"f:uid":{}}}},"f:spec":{"f:ports":{".":{},"k:{\"port\":7000,\"protocol\":\"TCP\"}":{".":{},"f:name":{},"f:port":{},"f:protocol":{},"f:targetPort":{}},"k:{\"port\":7001,\"protocol\":\"TCP\"}":{".":{},"f:name":{},"f:port":{},"f:protocol":{},"f:targetPort":{}},"k:{\"port\":7199,\"protocol\":\"TCP\"}":{".":{},"f:name":{},"f:port":{},"f:protocol":{},"f:targetPort":{}},"k:{\"port\":9042,\"protocol\":\"TCP\"}":{".":{},"f:name":{},"f:port":{},"f:protocol":{},"f:targetPort":{}},"k:{\"port\":9142,\"protocol\":\"TCP\"}":{".":{},"f:name":{},"f:port":{},"f:protocol":{},"f:targetPort":{}},"k:{\"port\":9160,\"protocol\":\"TCP\"}":{".":{},"f:name":{},"f:port":{},"f:protocol":{},"f:targetPort":{}},"k:{\"port\":10001,\"protocol\":\"TCP\"}":{".":{},"f:name":{},"f:port":{},"f:protocol":{},"f:targetPort":{}}},"f:publishNotReadyAddresses":{},"f:selector":{".":{},"f:statefulset.kubernetes.io/pod-name":{}},"f:sessionAffinity":{},"f:type":{}}}}]},"spec":{"ports":[{"name":"inter-node-communication","protocol":"TCP","port":7000,"targetPort":7000},{"name":"ssl-inter-node-communication","protocol":"TCP","port":7001,"targetPort":7001},{"name":"jmx-monitoring","protocol":"TCP","port":7199,"targetPort":7199},{"name":"cql","protocol":"TCP","port":9042,"targetPort":9042},{"name":"cql-ssl","protocol":"TCP","port":9142,"targetPort":9142},{"name":"agent-api","protocol":"TCP","port":10001,"targetPort":10001},{"name":"thrift","protocol":"TCP","port":9160,"targetPort":9160}],"selector":{"statefulset.kubernetes.io/pod-name":"scyllacluster-eu-central-1-a-0"},"clusterIP":"10.107.177.189","clusterIPs":["10.107.177.189"],"type":"ClusterIP","sessionAffinity":"None","publishNotReadyAddresses":true},"status":{"loadBalancer":{}}}],"_trace_id":"QpP6lMg7RR-Wsn7ws4I39A"}
{"L":"DEBUG","T":"2021-05-11T11:28:49.091Z","N":"cluster-controller","M":"Found nodes","len":1,"_trace_id":"QpP6lMg7RR-Wsn7ws4I39A"}
E0511 11:28:49.294558       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 524 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1b3b8a0, 0x2d56fa0)
	k8s.io/apimachinery@v0.20.4/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.20.4/pkg/util/runtime/runtime.go:48 +0x86
panic(0x1b3b8a0, 0x2d56fa0)
	runtime/panic.go:965 +0x1b9
github.com/scylladb/scylla-operator/pkg/controllers/cluster.(*ClusterReconciler).orphanedCleanup(0xc000a8c280, 0x20654b8, 0xc000826c60, 0xc00087ac80, 0xc0001f6a10, 0xc0001f6a10, 0x0)
	github.com/scylladb/scylla-operator/pkg/controllers/cluster/cleanup.go:90 +0x8a7
github.com/scylladb/scylla-operator/pkg/controllers/cluster.(*ClusterReconciler).cleanup(0xc000a8c280, 0x20654b8, 0xc000826c60, 0xc00087ac80, 0x20654b8, 0xc000826c60)
	github.com/scylladb/scylla-operator/pkg/controllers/cluster/cleanup.go:49 +0x87c
github.com/scylladb/scylla-operator/pkg/controllers/cluster.(*ClusterReconciler).sync(0xc000a8c280, 0xc00087ac80, 0x0, 0x0)
	github.com/scylladb/scylla-operator/pkg/controllers/cluster/sync.go:47 +0x465
github.com/scylladb/scylla-operator/pkg/controllers/cluster.(*ClusterReconciler).Reconcile(0xc000a8c280, 0x20654b8, 0xc000b15230, 0xc0009af98a, 0x6, 0xc0009af970, 0xd, 0xc000b15230, 0xc00003a000, 0x1c5eb80, ...)
	github.com/scylladb/scylla-operator/pkg/controllers/cluster/cluster_controller.go:122 +0xdcd
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000658280, 0x2065410, 0xc0002dd840, 0x1bf3f40, 0xc0004ff440)
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000658280, 0x2065410, 0xc0002dd840, 0x6f69737265566500)
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x2065410, 0xc0002dd840)
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216 +0x4a
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc00077df50)
	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000bf1f50, 0x202a540, 0xc000b151d0, 0xc0002dd801, 0xc00072c060)
	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00077df50, 0x3b9aca00, 0x0, 0x2c22746567222c01, 0xc00072c060)
	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x2065410, 0xc0002dd840, 0xc00059ea90, 0x3b9aca00, 0x0, 0x74696d696c223a01)
	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x2065410, 0xc0002dd840, 0xc00059ea90, 0x3b9aca00)
	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:99 +0x57
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:213 +0x40d
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x188a107]

goroutine 524 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.20.4/pkg/util/runtime/runtime.go:55 +0x109
panic(0x1b3b8a0, 0x2d56fa0)
	runtime/panic.go:965 +0x1b9
github.com/scylladb/scylla-operator/pkg/controllers/cluster.(*ClusterReconciler).orphanedCleanup(0xc000a8c280, 0x20654b8, 0xc000826c60, 0xc00087ac80, 0xc0001f6a10, 0xc0001f6a10, 0x0)
	github.com/scylladb/scylla-operator/pkg/controllers/cluster/cleanup.go:90 +0x8a7
github.com/scylladb/scylla-operator/pkg/controllers/cluster.(*ClusterReconciler).cleanup(0xc000a8c280, 0x20654b8, 0xc000826c60, 0xc00087ac80, 0x20654b8, 0xc000826c60)
	github.com/scylladb/scylla-operator/pkg/controllers/cluster/cleanup.go:49 +0x87c
github.com/scylladb/scylla-operator/pkg/controllers/cluster.(*ClusterReconciler).sync(0xc000a8c280, 0xc00087ac80, 0x0, 0x0)
	github.com/scylladb/scylla-operator/pkg/controllers/cluster/sync.go:47 +0x465
github.com/scylladb/scylla-operator/pkg/controllers/cluster.(*ClusterReconciler).Reconcile(0xc000a8c280, 0x20654b8, 0xc000b15230, 0xc0009af98a, 0x6, 0xc0009af970, 0xd, 0xc000b15230, 0xc00003a000, 0x1c5eb80, ...)
	github.com/scylladb/scylla-operator/pkg/controllers/cluster/cluster_controller.go:122 +0xdcd
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000658280, 0x2065410, 0xc0002dd840, 0x1bf3f40, 0xc0004ff440)
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000658280, 0x2065410, 0xc0002dd840, 0x6f69737265566500)
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x2065410, 0xc0002dd840)
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216 +0x4a
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc00077df50)
	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000bf1f50, 0x202a540, 0xc000b151d0, 0xc0002dd801, 0xc00072c060)
	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00077df50, 0x3b9aca00, 0x0, 0x2c22746567222c01, 0xc00072c060)
	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x2065410, 0xc0002dd840, 0xc00059ea90, 0x3b9aca00, 0x0, 0x74696d696c223a01)
	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x2065410, 0xc0002dd840, 0xc00059ea90, 0x3b9aca00)
	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:99 +0x57
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:213 +0x40d
```

**Environment:**
 - Platform: minikube
 - Kubernetes version: 1.20 
 - Scylla version: - 
 - Scylla-operator version: 1.2

--- a/pkg/controllers/cluster/cleanup.go
+++ b/pkg/controllers/cluster/cleanup.go
@@ -87,6 +87,11 @@ func (cc *ClusterReconciler) orphanedCleanup(ctx context.Context, c *scyllav1.Sc
 			return errors.Wrap(err, "get pv")
 		}
 
+		if pv.Spec.NodeAffinity == nil {
+			cc.Logger.Debug(ctx, "PV doesn't have node affinity", "pv", pv.Name)
+			continue
+		}
+
 		ns, err := nodeaffinity.NewNodeSelector(pv.Spec.NodeAffinity.Required)
 		if err != nil {
 			return errors.Wrap(err, "new node selector")

